### PR TITLE
Eliminate the use of containers/building import in pkg subdirs

### PR DESCRIPTION
**Pull request by rhatdan**
We want to shrink the size of the import when importing pkg from
buildah. This should help us shrink the size of the golang bindings
in podman.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL
@nalind @vrothberg @baude @TomSweeneyRedHat PTAL

**Review by vrothberg — COMMENTED** [review]
LGTM but I want a head nod from @nalind

File: define/pull.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	// should be pulled from a registry before the build proceeds.
+	PullAlways
+	// PullIfNewer is one of the values that BuilderOptions.PullPolicy
+	// can take, signalling that the source image should only be pulled
+	// from a registry if a local copy is not already present or if a
+	// newer version the image is present on the repository.
+	PullIfNewer
+	// PullNever is one of the values that BuilderOptions.PullPolicy can
+	// take, signalling that the source image should not be pulled from a
+	// registry if a local copy of it is not already present.
```
Comment (TomSweeneyRedHat) [inline define/pull.go:26]: It never pulls.  It get's the local copy or fails.  I don't know if you want to mention the failure or not here.

```suggestion
	// registry.
```

Comment (TomSweeneyRedHat) [reply define/pull.go:26]: and oops, I see that's pre-existing.

**Comment by TomSweeneyRedHat** [comment]
LGTM, but would like a @nalind head nod too if possible.

**Review by giuseppe — APPROVED** [review]
LGTM

**Pull request merged**